tfupdate/option.go

```diff
@@ -43,18 +51,35 @@ func NewOption(updateType string, name string, version string, platforms []strin
 
 		r, err := regexp.Compile(ignorePath)
 		if err != nil {
-			return Option{}, fmt.Errorf("faild to compile regexp for ignorePath: %s", err)
+			return Option{}, fmt.Errorf("failed to compile regexp for ignorePath: %s", err)
 		}
 		regexps = append(regexps, r)
 	}
 
+	var nameRegex *regexp.Regexp
+	validSourceMatchTypes := []string{"full", "regex"}
+	if !slices.Contains[string](validSourceMatchTypes, sourceMatchType) {
```

Comment:
Acceptance tests were failed.

```
2024/01/25 01:06:04 [INFO] CLI args: []string{"tfupdate", "provider", "null", "-v", "3.2.1", "-r", "./"}
2024/01/25 01:06:04 [INFO] Update provider null to 3.2.1
invalid sourceMatchType:  valid options [full,regex]
make: *** [Makefile:30: testacc-lock-simple] Error 1
Error: Process completed with exit code 2.
```

I completely forgot that the `Option` type is shared with subcommands other than `module`, but some flags only make sense for some subcommands. The `sourceMatchType` validation checks should be limited to only `updateType ==  "module"`.